src/java/org/apache/cassandra/cql3/CQLFragmentParser.java

```diff
@@ -43,6 +47,19 @@
         {
             return parseAnyUnhandled(parserFunction, input);
         }
+        catch (NullPointerException re)
+        {
+            logger.error("Failed parsing {}: [{}] reason: {}",
```

Comment:
is catching a NPE here necessary or could we move the error logging into the part where we catch the RuntimeException?